h2o-r/h2o-package/R/kvstore.R

```diff
@@ -266,3 +266,59 @@ h2o.download_pojo <- function(model, path="", getjar=NULL, get_jar=TRUE) {
 
   if( path!="") print( paste0("POJO written to: ", file.path) )
 }
+
+#'
+#' Download the model in MOJO format.
+#'
+#' @param model An H2OModel
+#' @param path The path where MOJO file should be saved. Saved to current directory by default.
+#' @param get_genmodel_jar If TRUE, then also download h2o-genmodel.jar and store it in folder ``path``.
+#' @return Name of the MOJO file written along with path to MOJO file.
+#'
+#' @examples
+#' \donttest{
+#' library(h2o)
+#' h <- h2o.init(nthreads=-1)
+#' fr <- as.h2o(iris)
+#' my_model <- h2o.gbm(x=1:4, y=5, training_frame=fr)
+#' h2o.download_mojo(my_model)  # save to the current working directory
+#' }
+#' @export
+h2o.download_mojo <- function(model, path=getwd(), get_genmodel_jar=FALSE) {
+
+  if(!(is.character(path))){
+    stop("The 'path' variable should be of type character")
+  }
+  if(!(is.logical(get_genmodel_jar))){
+    stop("The 'get_genmodel_jar' variable should be of type logical/boolean")
+  }
+
+  if(!(model@algorithm %in% c("drf","gbm"))){
+    stop("MOJOs are currently supported for Distributed Random Forest and Gradient Boosting Method models only.")
+  }
+
+  model_id <- model@model_id
+  .__curlError = FALSE
+  .__curlErrorMessage = ""
+
+  url = .h2o.calcBaseURL(h2oRestApiVersion = .h2o.__REST_API_VERSION, urlSuffix = paste0(.h2o.__MODELS,"/",model_id,"/mojo"))
+  tmp = tryCatch(getBinaryURL(url = url,
+                      useragent = R.version.string),
+               error = function(x) { .__curlError <<- TRUE; .__curlErrorMessage <<- x$message })
+  if (! .__curlError) {
+     mojo.path <- paste0(path,"/",model_id,".zip")
+     writeBin(tmp, mojo.path, useBytes = TRUE)
+  }
+  if (get_genmodel_jar) {
+    url = .h2o.calcBaseURL(h2oRestApiVersion = .h2o.__REST_API_VERSION, urlSuffix = "h2o-genmodel.jar")
+    tmp = tryCatch(getBinaryURL(url = url,
+                        useragent = R.version.string),
+                 error = function(x) { .__curlError <<- TRUE; .__curlErrorMessage <<- x$message })
+    if (! .__curlError) {
+      jar.path <- paste0(path, "/h2o-genmodel.jar")
+      writeBin(tmp, jar.path, useBytes = TRUE)
+    }
+  }
+
+  print( paste0("MOJO written to: ", mojo.path) )
```

Comment:
Thats probably an old commit. I return the following now:
`print( paste0("MOJO file name: ", model_id,".zip") )
 print( paste0("MOJO written to: ", mojo.path) )`